### DISCUSSION TITLE
Missing `bind_vector`?

### DISCUSSION THREAD

**Discussion by qnzhou** [wjakob/nanobind#26] [Q&A] [answered]
Hi Wenzel,

Thanks for another great library!  I noticed that `bind_vector`, `bind_map`, etc. are missing in nanobind.  Is there plan of adding them?  Thanks!

best,
James

**Comment by wjakob** [comment, marked as answer]
Hi James -- I have never used these myself and don't plan to add them. If you wish to propose a PR with this feature plus tests, and if it is self-contained (i.e., does not add any code to the core nanobind), then it would likely get in after review.

**Reply by qnzhou** [reply]
Thanks Wenzel!

**Reply by lpapp** [reply]
Have you ever got around to this @qnzhou ? I am also interested.

Alternatively, is there a way in nanobind to bind arrays/lists/etc?

**Reply by qnzhou** [reply]
I have not.  I found that exposing `std::vector` as `nanobind::tensor` a better option which avoids data copy overhead.  See the discussion about that in https://github.com/wjakob/nanobind/discussions/53.

**Reply by lpapp** [reply]
Is it really a drop-in replacement though? The name scares me a bit.

@wjakob would this be not a simple copy/paste/adjust from pybind11? At the moment, the lack of this holds us back from switch to pybind11 to nanobind as we do need to be able to bind vectors.
